### PR TITLE
fix logging on modern java

### PR DESCRIPTION
**Pull request by crafter23456**
# Description

Closes https://github.com/CobbleSword/NachoSpigot/issues/327

# How has this been tested?

watched into the log

# Additional Comments

```
[15:53:43] [Application-Thread/INFO]: Loading TacoSpigot config from taco.yml
[15:53:43] [Application-Thread/INFO]: Loading libraries, please wait...
```
these doesnt get send before, but now its directly at the beginning. might be because its on the application thread and not on the server thread like the others. should be with the nacho log and not directly at the first line but dunno how

# Checklist:

- [x] I have reviewed my code thoroughly.
- [x] I have tested my code.
- [x] My changes generate no new warnings

### CODE REVIEW

File: NachoSpigot-Server/src/main/java/net/minecraft/server/DedicatedServer.java [hunk 1]
```diff
@@ -130,9 +130,9 @@ public void run() {
         boolean eulaAgreed = Boolean.getBoolean( "com.mojang.eula.agree" );
         if ( eulaAgreed )
         {
-            System.err.println( "You have used the Spigot command line EULA agreement flag." );
-            System.err.println( "By using this setting you are indicating your agreement to Mojang's EULA (https://account.mojang.com/documents/minecraft_eula)." );
-            System.err.println( "If you do not agree to the above EULA please stop your server and remove this flag immediately." );
+            DedicatedServer.LOGGER.error( "You have used the Spigot command line EULA agreement flag." );
+            DedicatedServer.LOGGER.error( "By using this setting you are indicating your agreement to Mojang's EULA (https://account.mojang.com/documents/minecraft_eula)." );
+            DedicatedServer.LOGGER.error( "If you do not agree to the above EULA please stop your server and remove this flag immediately." );
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/minecraft/server/DedicatedServer.java:135]: Should be warnings

File: NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java [hunk 2]
```diff
@@ -31,9 +32,9 @@ public Nacho() {
         lagCompensator = new LagCompensator();
 
         if(NachoConfig.enableAntiCrash) {
-            System.out.println("[NS-AntiCrash] Activating Anti Crash.");
+            NachoLogger.LOGGER.info("[NS-AntiCrash] Activating Anti Crash.");
             this.packetListeners.add(new AntiCrash());
-            System.out.println("[NS-AntiCrash] Activated Anti Crash.");
+            NachoLogger.LOGGER.info("[NS-AntiCrash] Activated Anti Crash.");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java:37]: Remove unnecessary logging

File: NachoSpigot-Server/src/main/java/net/minecraft/server/Chunk.java [hunk 3]
```diff
@@ -958,9 +958,9 @@ public void a(BlockPosition blockposition, TileEntity tileentity) {
             this.tileEntities.remove(blockposition);
             // PaperSpigot end
         } else {
-            System.out.println("Attempted to place a tile entity (" + tileentity + ") at " + tileentity.position.getX() + "," + tileentity.position.getY() + "," + tileentity.position.getZ()
+            Chunk.c.error("Attempted to place a tile entity (" + tileentity + ") at " + tileentity.position.getX() + "," + tileentity.position.getY() + "," + tileentity.position.getZ()
                 + " (" + org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(getType(blockposition)) + ") where there was no entity tile!");
-            System.out.println("Chunk coordinates: " + (this.locX * 16) + "," + (this.locZ * 16));
+            Chunk.c.error("Chunk coordinates: " + (this.locX * 16) + "," + (this.locZ * 16));
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/minecraft/server/Chunk.java:963]: Should be warnings

File: NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/commons/NachoLogger.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package dev.cobblesword.nachospigot.commons;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class NachoLogger {
+    public static final Logger LOGGER = LogManager.getLogger(NachoLogger.class);
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/commons/NachoLogger.java:7]: Should be moved to Nacho.java

File: NachoSpigot-Server/src/main/java/net/minecraft/server/CrashReport.java [hunk 5]
```diff
@@ -266,7 +266,7 @@ public CrashReportSystemDetails a(String s, int i) {
             int k = astacktraceelement.length - j;
 
             if (k < 0) {
-                System.out.println("Negative index in crash report handler (" + astacktraceelement.length + "/" + j + ")");
+                CrashReport.a.error("Negative index in crash report handler (" + astacktraceelement.length + "/" + j + ")");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/minecraft/server/CrashReport.java:269]:
```suggestion
                CrashReport.a.warn("Negative index in crash report handler (" + astacktraceelement.length + "/" + j + ")");
```

File: NachoSpigot-Server/src/main/java/net/minecraft/server/PlayerConnection.java [hunk 6]
```diff
@@ -2175,7 +2175,7 @@ public void a(PacketPlayInCustomPayload packetplayincustompayload) {
                 }
             } catch (Throwable ex) {
                 this.isExploiter = true;
-                System.out.println(this.player.getName() + " has tried to crash the server...");
+                PlayerConnection.c.info(this.player.getName() + " has tried to crash the server...");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/minecraft/server/PlayerConnection.java:2178]:
```suggestion
```

File: NachoSpigot-Server/src/main/java/net/minecraft/server/World.java [hunk 7]
```diff
@@ -1680,7 +1678,7 @@ public void tickEntities() {
                 } catch (Throwable throwable1) {
                     // PaperSpigot start - Prevent tile entity and entity crashes
                     entity.tickTimer.stopTiming();
-                    System.err.println("Entity threw exception at " + entity.world.getWorld().getName() + ":" + entity.locX + "," + entity.locY + "," + entity.locZ);
+                    NachoLogger.LOGGER.info("Entity threw exception at " + entity.world.getWorld().getName() + ":" + entity.locX + "," + entity.locY + "," + entity.locZ);
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/minecraft/server/World.java:1681]:
```suggestion
                    NachoLogger.LOGGER.error("Entity threw exception at " + entity.world.getWorld().getName() + ":" + entity.locX + "," + entity.locY + "," + entity.locZ);
```

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java [hunk 8]
```diff
@@ -23,12 +25,12 @@ public static void main(String[] args) {
         System.setProperty("log4j2.formatMsgNoLookups", "true");
 
         try {
-            if(!SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_15)) {
-                System.err.println("It seems like you are not using Java 15!");
-                System.out.println("The use of Java 15 is strongly recommended.");
+            if(!SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_17)) {
+                NachoLogger.LOGGER.info("It seems like you are not using Java 17!");
+                NachoLogger.LOGGER.info("The use of Java 17 is strongly recommended.");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:30]: The recommended version is Java 11 or higher I believe

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java [hunk 9]
```diff
@@ -180,12 +182,12 @@ public static void main(String[] args) {
                 Logger.getLogger(Main.class.getName()).log(Level.SEVERE, null, ex);
             }
         } else if (options.has("v")) {
-            System.out.println(CraftServer.class.getPackage().getImplementationVersion());
+             NachoLogger.LOGGER.info(CraftServer.class.getPackage().getImplementationVersion());
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:185]:
```suggestion
            NachoLogger.LOGGER.info(CraftServer.class.getPackage().getImplementationVersion());
```

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java [hunk 10]
```diff
@@ -226,12 +228,12 @@ public static void main(String[] args) {
                 }
                 if ( Float.parseFloat( System.getProperty( "java.class.version" ) ) < 52 && maxPermGen < ( 128 << 10 ) ) // 128mb
                 {
-                    System.out.println( "Warning, your max perm gen size is not set or less than 128mb. It is recommended you restart Java with the following argument: -XX:MaxPermSize=128M" );
-                    System.out.println( "Please see http://www.spigotmc.org/wiki/changing-permgen-size/ for more details and more in-depth instructions." );
+                    NachoLogger.LOGGER.error( "Warning, your max perm gen size is not set or less than 128mb. It is recommended you restart Java with the following argument: -XX:MaxPermSize=128M" );
+                    NachoLogger.LOGGER.error( "Please see http://www.spigotmc.org/wiki/changing-permgen-size/ for more details and more in-depth instructions." );
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:232]: Warnings

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java [hunk 11]
```diff
@@ -226,12 +228,12 @@ public static void main(String[] args) {
...
                 }
                 // Spigot End
                 net.techcable.tacospigot.TacoSpigotConfig.init((File) options.valueOf("taco-settings")); // TacoSpigot - load config before we load libraries to allow access while loading
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:235]: The taco config should be loaded after the Loading libraries print

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java [hunk 12]
```diff
@@ -930,7 +932,7 @@ public static Cancellable handleStatisticsIncrease(EntityHuman entityHuman, net.
         } else {
             org.bukkit.Statistic stat = CraftStatistic.getBukkitStatistic(statistic);
             if (stat == null) {
-                System.err.println("Unhandled statistic: " + statistic);
+                NachoLogger.LOGGER.error("Unhandled statistic: " + statistic);
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java:935]:
```suggestion
                NachoLogger.LOGGER.warn("Unhandled statistic: " + statistic);
```

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/util/ShortConsoleLogFormatter.java [hunk 13]
```diff
@@ -24,7 +24,7 @@ public ShortConsoleLogFormatter(MinecraftServer server) {
                     date = (SimpleDateFormat) object;
                 }
             } catch (OptionException ex) {
-                System.err.println("Given date format is not valid. Falling back to default.");
+                MinecraftServer.LOGGER.error("Given date format is not valid. Falling back to default.");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/util/ShortConsoleLogFormatter.java:27]:
```suggestion
                MinecraftServer.LOGGER.warn("Given date format is not valid. Falling back to default.");
```

File: NachoSpigot-Server/src/main/java/org/spigotmc/SpigotConfig.java [hunk 14]
```diff
@@ -215,7 +215,7 @@ private static void bungee() {
         if ( version < 4 )
         {
             set( "settings.bungeecord", false );
-            System.out.println( "Oudated config, disabling BungeeCord support!" );
+            MinecraftServer.LOGGER.info("Oudated config, disabling BungeeCord support!");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/spigotmc/SpigotConfig.java:218]:
```suggestion
            MinecraftServer.LOGGER.warn("Oudated config, disabling BungeeCord support!");
```

File: NachoSpigot-Server/src/main/java/org/sugarcanemc/sugarcane/util/yaml/YamlCommenter.java [hunk 15]
```diff
@@ -61,7 +62,7 @@ public void saveComments(File file, boolean loadWorldConfig) throws IOException
             boolean noNewline = StringUtils.getIndentation(lines.get(line)) > StringUtils.getIndentation(lines.get(line - 1));
             if (line >= 0)
                 lines.add(line, (noNewline ?"":"\n") + prefix + _comment.getValue().replace("\n", "\n" + prefix));
-            else System.out.printf("Failed to find key %s in %s!", _comment.getKey(), file);
+            else NachoLogger.LOGGER.error("Failed to find key %s in %s!", _comment.getKey(), file);
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/sugarcanemc/sugarcane/util/yaml/YamlCommenter.java:65]:
```suggestion
            else NachoLogger.LOGGER.warn("Failed to find key %s in %s!", _comment.getKey(), file);
```

**Review by Elierrr — CHANGES_REQUESTED** [review]

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java (hunk 8 above)
Comment (crafter23456) [reply NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:30]: in the readme is Java 17 also recommended and with that PR i fixed the last known Java 11 > bug

File: NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java (hunk 2 above)
Comment (crafter23456) [reply NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java:37]: both or only activating?

Comment (Elierrr) [reply NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java:37]: > above that isnt anything

Yes, there’s the activating one

File: NachoSpigot-Server/src/main/java/net/techcable/tacospigot/ImmutableArrayMap.java [hunk 16]
```diff
@@ -50,11 +51,11 @@ public int compare(K o1, K o2) {
...
-            System.err.println("Less than 50% of values are sequential");
-            System.err.print(sequentalRangeSize);
-            System.err.print(" out of ");
-            System.err.println(size);
-            System.err.println("Expect reduced performance");
+            Nacho.LOGGER.error("Less than 50% of values are sequential");
+            Nacho.LOGGER.error(sequentalRangeSize);
+            Nacho.LOGGER.error(" out of ");
+            Nacho.LOGGER.error(size);
+            Nacho.LOGGER.error("Expect reduced performance");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/techcable/tacospigot/ImmutableArrayMap.java:58]: Should be warnings

**Review by Elierrr — CHANGES_REQUESTED** [review]

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java (hunk 8 above)
Comment (Elierrr) [reply NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java]: @Lucaskyy What do you think?

Comment (Lucaskyy) [reply NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java]: Performance wise, Java 17 is recommended. Some plugins just don't work on that version, sadly.

**Review by Elierrr — APPROVED** [review]

File: NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java [hunk 17]
```diff
@@ -74,4 +74,5 @@ public LagCompensator getLagCompensator() {
         return lagCompensator;
     }
 
+    public static final Logger LOGGER = LogManager.getLogger(Nacho.class);
```
Comment (hpfxd) [inline NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/Nacho.java:77]: maybe move this field to the top of the class above `INSTANCE`?

**Review by hpfxd — APPROVED** [review]

**Comment by Lucaskyy** [comment]
Could you please fix the conflicts?

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java [hunk 18]
```diff
@@ -226,13 +228,13 @@ public static void main(String[] args) {
...
-                    System.out.println( "Please see http://www.spigotmc.org/wiki/changing-permgen-size/ for more details and more in-depth instructions." );
+                    Nacho.LOGGER.warn( "Warning, your max perm gen size is not set or less than 128mb. It is recommended you restart Java with the following argument: -XX:MaxPermSize=128M" );
+                    Nacho.LOGGER.warn( "Please see http://www.spigotmc.org/wiki/changing-permgen-size/ for more details and more in-depth instructions." );
                 }
                 // Spigot End
-                net.techcable.tacospigot.TacoSpigotConfig.init((File) options.valueOf("taco-settings")); // TacoSpigot - load config before we load libraries to allow access while loading
-                System.out.println("Loading libraries, please wait...");
                 MinecraftServer.main(options);
+                Nacho.LOGGER.info("Loading libraries, please wait...");
+                net.techcable.tacospigot.TacoSpigotConfig.init((File) options.valueOf("taco-settings")); // TacoSpigot - load config before we load libraries to allow access while loading
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:237]:
```suggestion
                Nacho.LOGGER.info("Loading libraries, please wait...");
                net.techcable.tacospigot.TacoSpigotConfig.init((File) options.valueOf("taco-settings")); // TacoSpigot - load config before we load libraries to allow access while loading
                MinecraftServer.main(options);
```

**Review by Elierrr — CHANGES_REQUESTED** [review]

**Review by Elierrr — APPROVED** [review]

File: NachoSpigot-Server/src/main/java/net/techcable/tacospigot/ImmutableArrayMap.java [hunk 19]
```diff
@@ -50,11 +51,11 @@ public int compare(K o1, K o2) {
...
-            System.err.println("Less than 50% of values are sequential");
-            System.err.print(sequentalRangeSize);
-            System.err.print(" out of ");
-            System.err.println(size);
-            System.err.println("Expect reduced performance");
+            Nacho.LOGGER.warn("Less than 50% of values are sequential");
+            Nacho.LOGGER.warn(sequentalRangeSize);
+            Nacho.LOGGER.warn(" out of ");
+            Nacho.LOGGER.warn(size);
+            Nacho.LOGGER.warn("Expect reduced performance");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/net/techcable/tacospigot/ImmutableArrayMap.java:58]:
```suggestion
            Nacho.LOGGER.warn("Less than 50% of values are sequential; " + sequentalRangeSize + " out of " + size);
            Nacho.LOGGER.warn("Expect reduced performance");
```

File: NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java [hunk 20]
```diff
@@ -23,12 +25,12 @@ public static void main(String[] args) {
         System.setProperty("log4j2.formatMsgNoLookups", "true");
 
         try {
-            if(!SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_15)) {
-                System.err.println("It seems like you are not using Java 15!");
-                System.out.println("The use of Java 15 is strongly recommended.");
+            if(!SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_17)) {
+                Nacho.LOGGER.info("It seems like you are not using Java 17!");
+                Nacho.LOGGER.info("The use of Java 17 is strongly recommended.");
```
Comment (Elierrr) [inline NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:30]:
```suggestion
                Nacho.LOGGER.warn("It seems like you are not using Java 17!");
                Nacho.LOGGER.warn("The use of Java 17 is strongly recommended.");
```

Comment (crafter23456) [reply NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:30]: hm isnt info there enough

Comment (Elierrr) [reply NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/Main.java:30]: It should stand out imo

**Review by Elierrr — APPROVED** [review]

**Comment by crafter23456** [comment]
@Elierrr and further things I didnt do well, or can @Lucaskyy merge it?

**Comment by Elierrr** [comment]
Should be could to merge @Lucaskyy

**Comment by Lucaskyy** [comment]
Thank you for the contribution! :)

**Pull request merged**